lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -138,19 +220,25 @@ mlir::linalgx::mapToBRGEMMOp(RewriterBase &rewriter,
   if (!isa<linalg::GenericOp>(linalgOp))
     return rewriter.notifyMatchFailure(linalgOp, "expects a linalg.generic");
 
-  if (failed(checkStructure(linalgOp)))
+  bool isVNNILoop = isLinalgVNNILoop(linalgOp);
```

Comment:
I am a bit concerned of having all this checks here. Can we have a MapToBatchReduceVNNIGemm.cpp? I feel is already complicated the logic and adding VNNI on top makes it even harder to debug. What do you think?

Reply:
I could refactor to have a separate MapToBatchReduceVNNIGemm.cpp, only problem is lots of code is likely to end up duplicated.